iotilecore/iotile/core/hw/auth/auth_chain.py

```diff
@@ -58,7 +70,11 @@ def get_root_key(self, key_type, device_id):
             try:
                 return provider.get_root_key(key_type, device_id)
             except NotFoundError:
-                pass
+                try:
```

Comment:
Attempt to get the root key with the same provider if an alias for the device exists. (If using UUID failed, use MAC)